redis/src/main/scala/zio/redis/TestExecutor.scala

```diff
@@ -96,6 +96,281 @@ private[redis] final class TestExecutor private (
+        val unit = input.last.asString match {
+          case "m"  => Some(RadiusUnit.Meters)
+          case "km" => Some(RadiusUnit.Kilometers)
+          case "ft" => Some(RadiusUnit.Feet)
+          case "mi" => Some(RadiusUnit.Miles)
+          case _    => None
+        }
+        val members = (if (unit.isEmpty) input.tail else input.dropRight(1)).map(_.asString)
+
+        orWrongType(isSortedSet(key))(
+          for {
+            scoreMap <- sortedSets.getOrElse(key, Map.empty)
+            hashes    = members collect scoreMap
+          } yield
+            if (hashes.size == 2)
+              RespValue.bulkString(
+                Hash
+                  .distance(
+                    Hash.decodeHash(hashes(0).toLong),
+                    Hash.decodeHash(hashes(1).toLong),
+                    unit.getOrElse(RadiusUnit.Meters)
+                  )
+                  .toString
+              )
+            else RespValue.NullBulkString
+        )
+
+      case api.Geo.GeoHash =>
+        val key     = input.head.asString
+        val members = input.tail.map(_.asString)
+
+        orWrongType(isSortedSet(key))(
+          for {
+            scoreMap <- sortedSets.getOrElse(key, Map.empty)
+            hashes    = members.map(scoreMap.get)
+          } yield RespValue.Array(
+            hashes.map { hashOption =>
+              hashOption.map { hash =>
+                val geoHash = Hash.asGeoHash(hash.toLong)
+                RespValue.bulkString(geoHash)
+              }.getOrElse(RespValue.NullBulkString)
+            }
+          )
+        )
+
+      case api.Geo.GeoPos =>
+        val key     = input.head.asString
+        val members = input.tail.map(_.asString)
+
+        orWrongType(isSortedSet(key))(
+          for {
+            scoreMap <- sortedSets.getOrElse(key, Map.empty)
+            hashes    = members.map(scoreMap.get)
+          } yield RespValue.Array(
+            hashes.map { hashOption =>
+              hashOption.map { hash =>
+                val longLat = Hash.decodeHash(hash.toLong)
+                RespValue.array(
+                  RespValue.bulkString(longLat.longitude.toString),
+                  RespValue.bulkString(longLat.latitude.toString)
+                )
+              }.getOrElse(RespValue.NullArray)
+            }
+          )
+        )
+
+      case api.Geo.GeoRadius =>
+        val stringInput = input.map(_.asString)
+
+        val key    = stringInput.head
+        val center = LongLat(stringInput(1).toDouble, stringInput(2).toDouble)
+        val radius = stringInput(3).toDouble
+        val unit = stringInput(4) match {
+          case "m"  => RadiusUnit.Meters
+          case "km" => RadiusUnit.Kilometers
+          case "ft" => RadiusUnit.Feet
+          case "mi" => RadiusUnit.Miles
+        }
+        val withCoord = stringInput.find(_ == "WITHCOORD")
+        val withDist  = stringInput.find(_ == "WITHDIST")
+        val withHash  = stringInput.find(_ == "WITHHASH")
```

Comment:
`WithCoord`, `WithDist` and `WithHash` have stringify methods. Use it instead of raw values. Technically speaking it's less efficient but it reduces the amount of magic values roaming around the code.